providers/implementations/kdfs/hkdf.c.in

```diff
@@ -385,111 +389,96 @@ static int kdf_hkdf_set_ctx_params(void *vctx, const OSSL_PARAM params[])
 static const OSSL_PARAM *kdf_hkdf_settable_ctx_params(ossl_unused void *ctx,
                                                       ossl_unused void *provctx)
 {
-    static const OSSL_PARAM known_settable_ctx_params[] = {
-        HKDF_COMMON_SETTABLES,
-        OSSL_PARAM_octet_string(OSSL_KDF_PARAM_INFO, NULL, 0),
-        OSSL_FIPS_IND_SETTABLE_CTX_PARAM(OSSL_KDF_PARAM_FIPS_KEY_CHECK)
-        OSSL_PARAM_END
-    };
-    return known_settable_ctx_params;
+    return hkdf_set_ctx_params_list;
 }
 
-static int hkdf_common_get_ctx_params(KDF_HKDF *ctx, OSSL_PARAM params[])
+{- produce_param_decoder('hkdf_get_ctx_params',
+                         (['KDF_PARAM_SIZE',                    'size',   'size_t'],
+                          ['KDF_PARAM_DIGEST',                  'digest', 'utf8_string'],
+                          ['KDF_PARAM_MODE',                    'mode',   'utf8_string'],
```

Comment:
Does the generated output for this multiple use of 'mode' work?

Reply:
They collapse into one path in the TRIE, so either flavour is found.  It's up to the calling code to deal with the different types.